botocore/session.py

```diff
@@ -796,6 +799,37 @@ def create_client(self, service_name, region_name=None, api_version=None,
             client_config=config, api_version=api_version)
         return client
 
+    def get_available_partitions(self):
+        """Lists the available partitions found on disk
+
+        :rtype: list
+        :return: Returns a list of partition names (e.g., ["aws", "aws-cn"])
+        """
+        resolver = self.get_component('endpoint_resolver')
+        return resolver.get_available_partitions()
+
+    def get_available_regions(self, service_name, partition_name='aws',
```

Comment:
I'm using this in a script I'm working on and I noticed that the `service_name` is actually the endpoint prefix.  What do you think about getting this updated to use botocore's service name?

The code I had was:

```
for service_name in session.get_available_services():
    for region_name in session.get_available_regions(service_name):
        # ...
```

Given other session methods that take service name use botocore's service names, it seems like `get_available_regions` should probably also use the same names:

```
>>> s = botocore.session.get_session()
>>> s.create_client('elb')
<botocore.client.ElasticLoadBalancing object at 0x10f82df10>
>>> s.get_service_model('elb')
<botocore.model.ServiceModel object at 0x10f780e10>
>>> s.get_available_regions('elb')
[]
>>> s.get_available_regions('elasticloadbalancing')
[u'us-east-1', u'us-west-1', u'us-west-2', ...]
```